bugsnag-plugin-android-anr/src/main/jni/anr_google.c

```diff
@@ -0,0 +1,96 @@
+#include "anr_google.h"
+
+#include <ctype.h>
+#include <dirent.h>
+#include <inttypes.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/syscall.h>
+#include <unistd.h>
+
+static pid_t process_id;
+static pid_t google_thread_id;
+
+static void trim_string_right(char* start) {
+  char* end = start + strlen(start);
+  while(start < end && isspace(*(end - 1))) {
+    end--;
+  }
+  *end = (char)0;
+}
+
+static bool is_thread_named_signal_catcher(pid_t tid) {
+  static const char*const SIGNAL_CATCHER_THREAD_NAME = "Signal Catcher";
+
+  bool success = false;
+  char buff[256];
+
+  snprintf(buff, sizeof(buff), "/proc/%d/comm", tid);
+  FILE* fp = fopen(buff, "r");
+  if(fp == NULL) {
+    return false;
+  }
+
+  if(fgets(buff, sizeof(buff), fp) != NULL) {
+    trim_string_right(buff);
+    success = strcmp(buff, SIGNAL_CATCHER_THREAD_NAME) == 0;
```

Comment:
Would `strncmp` work here instead for comparing the two strings? I don't quite follow why the string needs whitespace trimmed off - is that returned as part of the `fgets` call?